base/bucket_n1ql.go

```diff
@@ -126,48 +126,43 @@ func (bucket *CouchbaseBucketGoCB) BuildIndexes(indexNames []string) error {
 	return err
 }
 
-// If any of the provided indexes are in a pending state, attempt to rebuild those indexes.  On error, requery for indexes in pending
-// state and reattempts, up to max retries
-func (bucket *CouchbaseBucketGoCB) BuildPendingIndexes(indexSet []string, maxRetries int) error {
+// Issues a build command for any deferred sync gateway indexes associated with the bucket.  Waits for
+func (bucket *CouchbaseBucketGoCB) BuildDeferredIndexes(indexSet []string) error {
 
 	if len(indexSet) == 0 {
 		return nil
 	}
 
 	var buildErr error
-	for i := 1; i <= maxRetries; i++ {
-		statement := fmt.Sprintf("SELECT indexes.name, indexes.state from system:indexes WHERE indexes.keyspace_id = '%s' and indexes.name in [%s]", bucket.GetName(), StringSliceToN1QLArray(indexSet, "'"))
-		n1qlQuery := gocb.NewN1qlQuery(statement)
-		results, err := bucket.ExecuteN1qlQuery(n1qlQuery, nil)
-		if err != nil {
-			return err
-		}
-		pendingIndexes := make([]string, 0)
-		var indexInfo gocb.IndexInfo
-		for results.Next(&indexInfo) {
-			if indexInfo.State == IndexStatePending || indexInfo.State == IndexStateDeferred {
-				pendingIndexes = append(pendingIndexes, indexInfo.Name)
-			}
-		}
-		closeErr := results.Close()
-		if closeErr != nil {
-			return closeErr
-		}
 
-		if len(pendingIndexes) == 0 {
-			return nil
-		}
-		Infof(KeyQuery, "Building deferred indexes: %v", pendingIndexes)
-		buildErr = bucket.BuildIndexes(pendingIndexes)
-		// If build is successful, return
-		if buildErr == nil {
-			return nil
+	// Only build indexes that are in deferred state.  Query system:indexes to validate the provided set of indexes
+	statement := fmt.Sprintf("SELECT indexes.name, indexes.state from system:indexes WHERE indexes.keyspace_id = '%s' and indexes.name in [%s]", bucket.GetName(), StringSliceToN1QLArray(indexSet, "'"))
```

Comment:
I think this would be slightly easier to read if it was formatted like:

```
SELECT indexes.name, indexes.state FROM 
    system:indexes WHERE 
            indexes.keyspace_id = '%s' AND 
            indexes.name in [%s]", 
bucket.GetName(), StringSliceToN1QLArray(indexSet, "'")
```

(or something along those lines, where each chunk of the statement was on it's own line, with keywords in caps)